instrumentation/faraday/lib/opentelemetry/instrumentation/faraday/middlewares/tracer_middleware.rb

```diff
@@ -44,9 +44,9 @@ def call(env)
           def span_creation_attributes(http_method:, url:)
             instrumentation_attrs = {
               'http.method' => http_method,
-              'http.url' => OpenTelemetry::Common::Utilities.cleanse_url(url.to_s),
-              'net.peer.name' => url.host
+              'http.url' => OpenTelemetry::Common::Utilities.cleanse_url(url.to_s)
             }
+            instrumentation_attrs['net.peer.name'] = url.host if url.host
```

Comment:
Rather than check for individual attributes and conditionally set them, what do you think about using `instrumentation_attrs.compact!` before returning the hash?

That will remove any entries with `nil` values in the hash:

```console
irb(main):003> h = { "a" => 1, "b" => nil  }
=> {"a"=>1, "b"=>nil}
irb(main):004> h.compact!
=> {"a"=>1}
```

Reply:
I agree that it is more elegant to use `compact!` but when I ran benchmark, it performed slower than using `if`: 

```rb
def use_compact(b)
  h = { "a" => 1, "b" => b }
  h.compact!
end

def use_if(b)
  h = { "a" => 1 }
  h["b"] = b if b
end

Benchmark.ips do |x|
  x.report("compact! with nil") { use_compact(nil) }
  x.report("if with nil") { use_if(nil) }
  x.report("compact! with value") { use_compact(3) }
  x.report("if with value") { use_if(3) }

  x.compare!
end
```

```
Comparison:
         if with nil: 12207916.0 i/s
       if with value:  8454274.8 i/s - 1.44x  slower
   compact! with nil:  6958723.8 i/s - 1.75x  slower
 compact! with value:  6357076.8 i/s - 1.92x  slower
```

I don't mind switching to `compact!` but I feel current implementation is better... What do you think?

Reply:
Thanks for doing that benchmark! 

In that case I think we are good to go. Thank you again for this contribution.